lib/ansible/galaxy/api.py

```diff
@@ -696,17 +697,26 @@ def wait_import_task(self, task_id, timeout=0):
         start = time.time()
         wait = 2
 
+        token_expired = False
         while timeout == 0 or (time.time() - start) < timeout:
             try:
                 data = self._call_galaxy(full_url, method='GET', auth_required=True,
                                          error_context_msg='Error when getting import task results at %s' % full_url)
             except GalaxyError as e:
-                if e.http_code != 404:
+                if e.http_code == 401 and self.token and isinstance(self.token, KeycloakToken):
```

Comment:
Checking for `self.token` is probably redundant. It seems to be an `is None` check, but `KeycloakToken` doesn't even implement `__bool__` so that's kinda pointless. Or is it some sort of optimization?
```suggestion
                if e.http_code == 401 and isinstance(self.token, KeycloakToken):
```

Reply:
Yeah, it wasn't necessary, I just saw that `self.token` is initially None. But that will still work with the isinstance check.

Reply:
exactly